src/test/integration/users/User.sol

```diff
@@ -0,0 +1,257 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+import "forge-std/Test.sol";
+
+import "src/contracts/core/DelegationManager.sol";
+import "src/contracts/core/StrategyManager.sol";
+import "src/contracts/pods/EigenPodManager.sol";
+
+import "src/contracts/interfaces/IDelegationManager.sol";
+import "src/contracts/interfaces/IStrategy.sol";
+
+import "src/test/integration/Global.t.sol";
+
+contract User is Test {
+
+    Vm cheats = Vm(HEVM_ADDRESS);
+
+    DelegationManager delegationManager;
+    StrategyManager strategyManager;
+    EigenPodManager eigenPodManager;
+
+    IStrategy[] depositedStrategies;
+
+    Global global;
+
+    IStrategy constant BEACONCHAIN_ETH_STRAT = IStrategy(0xbeaC0eeEeeeeEEeEeEEEEeeEEeEeeeEeeEEBEaC0);
+
+    constructor(
+        DelegationManager _delegationManager,
+        StrategyManager _strategyManager,
+        EigenPodManager _eigenPodManager,
+        Global _global
+    ) {
+        delegationManager = _delegationManager;
+        strategyManager = _strategyManager;
+        eigenPodManager = _eigenPodManager;
+        global = _global;
+    }
```

Comment:
should we perhaps just have the User contract read this info from its deployer (msg.sender), so the constructor doesn't need any inputs?

Reply:
+1 for this - much cleaner

Reply:
would require a circular import, no?

Reply:
you should be able to make an interface and then cast the caller to that interface
something like 
```
interface IUserDeployer {
function delegationManager() external view returns (DelegationManager);
...
}
```

Reply:
[c99e847](https://github.com/Layr-Labs/eigenlayer-contracts/pull/335/commits/c99e847709852d7246c73b7d72d44bba368b760e)